pay-api/src/pay_api/services/eft_short_name_summaries.py

```diff
@@ -0,0 +1,136 @@
+
+        summary_list = [EFTSummarySchema.from_row(short_name_summary) for short_name_summary in pagination.items]
+        converter = Converter()
+        summary_list = converter.unstructure(summary_list)
+
+        current_app.logger.debug('>search')
+        return {
+            'state_total': search_count,
+            'page': search_criteria.page,
+            'limit': search_criteria.limit,
+            'items': summary_list,
+            'total': pagination.total
+        }
+
+    @staticmethod
+    def get_last_payment_received_query():
+        """Query for EFT most recent transaction deposit date."""
+        return db.session.query(
+            EFTTransactionModel.deposit_date,
+            EFTTransactionModel.short_name_id,
+            func.row_number().over(partition_by=EFTTransactionModel.short_name_id,
+                                   order_by=[EFTTransactionModel.deposit_date.desc(),
+                                             EFTTransactionModel.id]).label('rn')
+        ).filter(and_(EFTTransactionModel.short_name_id.isnot(None),
+                      EFTTransactionModel.status_code == EFTProcessStatus.COMPLETED.value)) \
+            .filter(EFTTransactionModel.line_type == EFTFileLineType.TRANSACTION.value)
+
+    @staticmethod
+    def get_linked_count_query():
+        """Query for EFT linked account count."""
+        # pylint: disable=not-callable
+        return (db.session.query(EFTShortnameLinksModel.eft_short_name_id,
+                                 func.count(
+                                     EFTShortnameLinksModel.eft_short_name_id
+                                 ).label('count'))
+                .filter(EFTShortnameLinksModel.status_code.in_([EFTShortnameStatus.PENDING.value,
+                                                                EFTShortnameStatus.LINKED.value]))
+                .group_by(EFTShortnameLinksModel.eft_short_name_id))
+
+    @staticmethod
+    def get_remaining_credit_query():
+        """Query for EFT remaining credit amount."""
+        return (db.session.query(EFTCreditModel.short_name_id,
+                                 func.sum(EFTCreditModel.remaining_amount).label('total'))
+                .group_by(EFTCreditModel.short_name_id))
+
+    @staticmethod
+    def get_search_count():
+        """Get a total count of short name summary results."""
+        current_app.logger.debug('<get_search_count')
+
+        count_query = (db.session.query(EFTShortnameModel.id)
+                       .group_by(EFTShortnameModel.id).with_entities(EFTShortnameModel.id))
+
+        current_app.logger.debug('>get_search_count')
+        return count_query.count()
+
+    @classmethod
+    def get_search_query(cls, search_criteria: EFTShortnamesSearch):
+        """Query for short names based on search criteria."""
+        linked_account_subquery = cls.get_linked_count_query().subquery()
+        credit_remaining_subquery = cls.get_remaining_credit_query().subquery()
```

Comment:
Is it worth moving this into the relationship so we get it for free?

Reply:
Perhaps something like this? 

https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html

So we don't need to use the service to query the credits etc.

Reply:
EG `relationship(Child, primaryjoin=Child.parent_id == Parent.id)` not sure if that's possible